### PR TITLE
fix: escape user id for firelfy store

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes #mf-4068

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-providers/src/StringStorage/index.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ interface ResponseData {
     message: string
     metaData: StringStorageBaseAPI.MetaData
 }
-const genKey = (namespace: string, userId: string) => `${namespace}-${userId}`
+const genKey = (namespace: string, userId: string) => `${namespace}-${userId.replace('.', '%2E')}`
```
Comment (guanbinrui) [inline packages/web3-providers/src/StringStorage/index.ts]: To use escape stead?

Comment (lelenei) [reply packages/web3-providers/src/StringStorage/index.ts]: period is a reserved character

Comment (guanbinrui) [reply packages/web3-providers/src/StringStorage/index.ts]: What if there are multiple .?

File: packages/web3-providers/src/StringStorage/index.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ interface ResponseData {
     message: string
     metaData: StringStorageBaseAPI.MetaData
 }
-const genKey = (namespace: string, userId: string) => `${namespace}-${userId}`
+const genKey = (namespace: string, userId: string) => `${namespace}-${userId.replace(/\./g, '%2E')}`
```
Comment (Jack-Works) [inline packages/web3-providers/src/StringStorage/index.ts:12]: or maybe `encodeURIComponent`?

Comment (lelenei) [reply packages/web3-providers/src/StringStorage/index.ts:12]: period is a reserved character

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**